src/app/screens/nftDetail/useNftDetail.ts

```diff
@@ -0,0 +1,111 @@
+import useStacksCollectibles from '@hooks/queries/useStacksCollectibles';
+import useNftDataSelector from '@hooks/stores/useNftDataSelector';
+import useNftDataReducer from '@hooks/stores/useNftReducer';
+import useResetUserFlow from '@hooks/useResetUserFlow';
+import useWalletSelector from '@hooks/useWalletSelector';
+import { getNftDetail, NftDetailResponse } from '@secretkeylabs/xverse-core';
+import { NftData } from '@secretkeylabs/xverse-core/types/api/stacks/assets';
+import { useMutation } from '@tanstack/react-query';
+import { GAMMA_URL } from '@utils/constants';
+import { getExplorerUrl, isLedgerAccount } from '@utils/helper';
+import { useEffect, useMemo, useState } from 'react';
+import { useNavigate, useParams } from 'react-router-dom';
+
+export default function useNftDetail() {
+  const navigate = useNavigate();
+  const { stxAddress, selectedAccount } = useWalletSelector();
+  const stacksNftsQuery = useStacksCollectibles();
+
+  const { id } = useParams();
+  const nftIdDetails = id!.split('::');
+  const collectionInfo = stacksNftsQuery.data?.pages
+    ?.map((page) => page?.results)
+    .flat()
+    .find((collection) => collection.collection_id === nftIdDetails[0]);
```

Comment:
```suggestion
  const { id } = useParams();
  const [collectionId] = id?.split('::');
  const collectionInfo = stacksNftsQuery.data?.pages
    ?.map((page) => page?.results)
    .flat()
    .find((collection) => collection.collection_id === collectionId);
```

if the query param isn't set, do not want the page to crash. prefer to degrade to a page with empty data